extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/FlatClassLoaderService.java

```diff
@@ -98,6 +98,20 @@ public InputStream locateResourceStream(String name) {
         throw new AssertionFailure("Not implemented! generateProxy(InvocationHandler handler, Class... interfaces)");
     }
 
+    @Override
+    public Package packageForNameOrNull(String packageName) {
+        try {
+            Class<?> aClass = Class.forName(packageName + ".package-info", false, getClassLoader());
```

Comment:
I'm not sure these will be included in the native image? Given it's a runtime class, I suppose it can be called at runtime?

For JAXB, I had to add this: https://github.com/quarkusio/quarkus/blob/master/extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java#L152 .

I don't know if you're having this new method for a similar purpose.

Reply:
No we don't need to access these at runtime, the purpose is for package-level annotations to be discoverable during metadata scanning when not using Jandex. I even doubt it's at all useful for Quarkus.

To be clear, this is code that previously was handled by HCANN, I moved the responsibility to the ClassLoaderService - this method is just doing exactly the same as it was done before.

Reply:
OK, if it's used when we build the metadata at build time, it's all good.